### PR TITLE
Only set node-port for one service port in Kind and Minikube

### PR DESCRIPTION
**Pull request by Sgitario**
The rules are:
- If there is only one port, we use it
- If a user sets a node-port in one and only one of the ports in `dekorate.[kind or minikube or kubernetes].ports, we use it.
- Otherwise, we try to use the named port "http" -> this is for backward compatibility. 

Fix https://github.com/dekorateio/dekorate/issues/1120

### CODE REVIEW

**Review by iocanel — APPROVED** [review]
I relluctantly approve as I find it slightly hard to follow.
More comments etc could possibly help.

**Comment by Sgitario** [comment]
> I relluctantly approve as I find it slightly hard to follow. More comments etc could possibly help.

PR updated according to your comments.
